src/Elasticsearch/Test/Product/ElasticsearchProductTest.php

```diff
@@ -3106,6 +3143,50 @@ public function variantListingConfigProvider(): iterable
         yield 'Should not index main product when displayParent is false' => ['productIds' => 'variant-2', 'expected' => 2];
     }
 
+    /**
+     * @depends testIndexing
+     */
+    public function testRangeAggregation(IdsCollection $data): void
+    {
+        $rangesDefinition = [
+            [],
+            ['key' => 'all'],
+            ['key' => 'custom_key', 'from' => 0, 'to' => 200],
+            ['to' => 100],
+            ['from' => 100, 'to' => 160],
+            ['from' => 200, 'to' => 500],
+            ['to' => 500],
+        ];
+
+        $rangesExpectedResult = [
+            '*-*' => 7,
+            'all' => 7,
+            'custom_key' => 3,
+            '*-100' => 1,
+            '100-160' => 2,
+            '200-500' => 4,
+            '*-500' => 7,
+        ];
```

Comment:
That looks like an inlined data provider. Can you add an actual dataProvider like the method above